### ISSUE TITLE
[$125] TalkBack: Many Pages: Cannot activate the button 'Send' cannot be activated

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77460]
## Action Performed:
Precondition: User is logged in

1. Using TalkBack on Android, open the app
2. Swipe right to focus any chat and double-tap to activate it
3. Swipe right to focus the text field and double-tap to activate it
4. Type in any text
5. Swipe right to focus the Send button and double-tap to activate it
6. Note if the Send button is operable and if the message is sent

Other Occurrences:
1. On New Chat
2. On Group Chat

## Expected Result:
The user should be able to activate the element by double tapping with an enabled screen reader.

## Actual Result:
The Send button is announced with a hint 'double tap to activate,' but the button cannot be activated by double tapping.

## Area issue was found in:
Chat

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure the button is correctly implemented and can be activated with an enabled screen reader.
https://appt.org/en/docs/react-native/samples/accessibility-role
https://reactnative.dev/docs/accessibility#accessibilityrole

## Environment:
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

<img width="1080" height="2195" alt="Image" src="https://github.com/user-attachments/assets/54755468-dacb-4841-813d-2e61d8eb5a05" />

https://github.com/user-attachments/assets/8e86b943-fd90-4e34-8c26-530418e9be1c

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7009080
**Issue reported by:** NA
**Slack conversation:** NA
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022009685327077697002</li>
        <li>Upwork Job ID: 2009685327077697002</li>
        <li>Last Price Increase: 2026-01-09</li>
<li>Automatic offers: </li>
<ul>
<li>linhvovan29546 | Contributor | 110142504</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @carlosmiceli</details>

**Comment by rushatgabhane** [comment]
the correct amount for this issue is $125

**Comment by QichenZhu** [comment]
@TaduJR, thanks for the proposal. Could you look into why we are using gesture? I don't think using both gesture and `onPress` is a good idea since it triggers `handleSendMessage()` twice when TalkBack is off.

**Comment by linhvovan29546** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
TalkBack: Many Pages: Cannot activate the button 'Send' cannot be activated 
### What is the root cause of that problem?
In the `SendButton` component, `handleSendMessage` is triggered via a `GestureDetector` when the user taps the button:
https://github.com/Expensify/App/blob/e94f05d8f5b32c3f916c0d4d159536bae4f120bd/src/pages/home/report/ReportActionCompose/SendButton.tsx#L41-L49
Because of this, `TalkBack` users cannot activate the `Send` button. Accessibility actions are only supported on Pressable components (PressableWithFeedback), not work correct on GestureDetector.
### What changes do you think we should make in order to solve the problem?
We can remove the `GestureDetector` and rely on the `onPress` handler of `PressableWithFeedback` instead:
https://github.com/Expensify/App/blob/e94f05d8f5b32c3f916c0d4d159536bae4f120bd/src/pages/home/report/ReportActionCompose/SendButton.tsx#L49

```
           <PressableWithFeedback
                            onPress={handleSendMessage}
```

The `GestureDetector` was originally introduced to fix an issue where the text was not cleared after submitting a message (see https://github.com/Expensify/App/issues/10148
) by sending the message on the UI thread. However, with the current implementation, taps on `GestureDetector` now run on the [JS thread](https://github.com/Expensify/App/blob/e94f05d8f5b32c3f916c0d4d159536bae4f120bd/src/pages/home/report/ReportActionCompose/SendButton.tsx#L33), and `handleSendMessage` has been updated to use `scheduleOnUI`. As a result, the original issue no longer reproduces, making the `GestureDetector` unnecessary.

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by QichenZhu** [comment]
> The issue also reproduces without removing GestureDetector but much less frequently.

@linhvovan29546, do you mean https://github.com/Expensify/App/issues/10148 is still reproducible currently? If so, should we reopen https://github.com/Expensify/App/issues/10148 and put this on hold?

**Comment by linhvovan29546** [comment]
This is hard to reproduce, and it’s slightly different as you can see in the screenshot, the text is not fully cleared at the 00:06. 

https://github.com/user-attachments/assets/59490f87-92ed-4beb-94e3-c04d7347bf89

**Comment by linhvovan29546** [comment]
If I remove `GestureDetector`, then https://github.com/Expensify/App/issues/10148 is easy to reproduce by following the steps in the OP

**Comment by linhvovan29546** [comment]
Update: I’m debugging `react-native-gesture-handler` on Android with `TalkBack` enabled to identify the root cause. I’ll post an update tomorrow.

**Comment by twisterdotcom** [comment]
@QichenZhu has submitted payment for this report on ND [here](https://staging.new.expensify.com/search?q=sortBy:date%20sortOrder:desc%20type:expense%20groupBy:reports%20reportID:R00K5xhbKo2Q%20policyID:9132779747441AFB).

SO: https://stackoverflowteams.com/c/expensify/questions/7582
- [x] Vendor found in [VML](https://docs.google.com/spreadsheets/d/1Fs-umMmHhCCMzxiczyNydl9Rd6jdO0pQskYA3RvNWQc/edit?gid=1740618654#gid=1740618654): True. Status: Active
- [x] Amount requested: $125.00 (Matches the payment summary here: https://github.com/Expensify/App/issues/77460#issuecomment-3908222639.)
- [x] I, @twisterdotcom am not already the BZ assignee. I have approved this payment.

<details><summary>Zapier Logs</summary>
<b>Run ID:</b> <a href="https://zapier.com/editor/311565481/run/00040eee-23e9-a509-bf8f-a22434595414/run-details">00040eee-23e9-a509-bf8f-a22434595414</a>
</details>
